### PR TITLE
Create sms email form html template

### PR DESCRIPTION
**Pull request by Koji23**
https://user-images.githubusercontent.com/10385763/177871650-e4d829e4-d09a-4dbd-88ec-9d0b94a49260.mov

### CODE REVIEW

**Comment by gardenFiend138** [comment]
Can we also add Charles as a reviewer to this? Also have we tested this across devices/ screen sizes to see how it looks?

**Comment by Koji23** [comment]
> Can we also add Charles and Olive as reviewers to this? Also have we tested this across devices/ screen sizes/ in landscape mode to see how it looks?

yup, I've invited them to the project

tested this on a few different devices including my iphone, android sim, and a tablet android sim (shown below)

Screen Shot 2022-07-07 at 5 21 09 PM

Also no this does not look good in landscape mode. I considered investing more time to do that but if we did we'd probably need to do that across every template which is an additional time investment and currently @rubychilds is suggesting we can just specify that this is a portrait only template (https://onesignal.slack.com/archives/C01T6B1KDEG/p1656703545310619?thread_ts=1656612136.984279&cid=C01T6B1KDEG)

**Comment by Koji23** [comment]
I got some design feedback from Olive and Ruby here https://onesignal.slack.com/archives/C01T6B1KDEG/p1657232145299789?thread_ts=1657227737.943509&cid=C01T6B1KDEG

**Comment by Koji23** [comment]
@gardenFiend138  sidenote: I think we will want to start injecting the app-id into the iam's document scope so that it's automatically available there (similar to how we do for iam-id) this will make it easier to use the public api and cleaner in this template

**Comment by Koji23** [comment]
TODO (now)
- [ ] add notch detection to top level readme with link example
ie.
```
## Safe Area (notch) Detection
Certain mobile devices may have safe areas where operating system icons or a camera notch might exist. Our Onesignal SDK detects these areas and exposes them to the iam's HTML document via `safe-area-inset-*` CSS variables :point_down:

https://github.com/OneSignalDevelopers/iam-html-templates/blob/00e30f0240e22f2e72ae807ac9274c3d0ab97bc8/sms_email_form/index.html#L21-L31
```

File: sms_email_form/index.html [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+            if (emailAddress) {
+              createEmailPlayer(emailAddress);
+            }
+
+            if (phoneNumber) {
+              createSmsPlayer(phoneNumber);
+            }
+            OneSignalIamApi.close(e);
+            return false;
```
Comment (gardenFiend138) [inline sms_email_form/index.html:283]: why do we return false here?

File: sms_email_form/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# SMS + Email Form Template
+
+This HTML template will let you request email addresses and phone numbers from users and create new SMS and Email players out of them. You will need to add your Onesignal App ID into this template at the appropriate place.
+
+###### SMS Email form on Android
+![Screenshot](readme_assets/sms_email_form.png)
+
+## Safe Area (notch) Detection
+Certain mobile devices may have safe areas where operating system icons or a camera notch might exist. Our Onesignal SDK detects these areas and exposes them to the iam's HTML document via `safe-area-inset-*` CSS variables :point_down:
```
Comment (gardenFiend138) [inline sms_email_form/README.md:9]: do we have these variables exposed anywhere else in the documentation? i.e. in the custom html docs in our main docs?

**Review by gardenFiend138 — COMMENTED** [review]
so far lgtm, holding off since you left a comment of some TODOs in the pr comments

File: README.md [hunk 3]
```diff
@@ -1,2 +1,10 @@
 # iam html templates
- HTML templates for in-app messaging 
+ HTML templates for in-app messaging
+
+## In-app Messaging Templates
+* [Leads (SMS + Email)](./sms_email_form/README.md)
+* [Survey (Ranking 1-5)](#/README.md)
+* [Carousel](#/README.md)
+* [Responsive](#/README.md)
+* [Promo Wheel](#/README.md)
```
[the remark this answers is not in the document]
Comment (Koji23) [reply README.md]: sorry didn't see this comment but on a separate PR this was already merged to main

**Review by iAmWillShepherd — APPROVED** [review]
Token approval

File: sms_email_form/index.html (hunk 1 above)
Comment (Koji23) [reply sms_email_form/index.html]: done

**Pull request merged**